# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/remoteworkers/PRemoteWorkerDebugger.ipdl
# Commit: 2596059bf325
# Full Hash: 2596059bf3254077a58f6bdda57796b3b20e00a5
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2025-03-18 21:32:34
# Regressor Bug: 1899503
# File Overlap Count: 2
# Description:
#   Bug 1899503 - P1 New top-level IPC PRemoteWorkerDebuggerManager and PRemoteWorkerDebugger to support nsIWorkerDebugger interface for Workers in the parent process. r=asuth
#   
#   New IPC PRemoteWorkerDebuggerManager and PRemoteWorkerDebugger are introduced to support remote debugger of Workers on the parent thread.
#   
#   PRemoteWorkerDebuggerManager is IPC between the parent process main thread and the content process WorkerLauncher thread.
# ==============================================================================

diff -r 2cd7a5fae8ef -r 2596059bf325 dom/workers/remoteworkers/PRemoteWorkerDebugger.ipdl
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/workers/remoteworkers/PRemoteWorkerDebugger.ipdl	Tue Mar 18 12:10:53 2025 +0000
@@ -0,0 +1,33 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+include RemoteWorkerTypes;
+
+namespace mozilla {
+namespace dom{
+
+[ChildProc=anydom]
+protocol PRemoteWorkerDebugger
+{
+parent:
+  async Unregister();
+
+  async PostMessageToDebugger(nsString aMessage);
+  async ReportErrorToDebugger(RemoteWorkerDebuggerErrorInfo aErrorInfo);
+
+  async SetAsInitialized();
+  async SetAsClosed();
+
+  // IPC for updating SharedWorker's WindowIDs 
+  async AddWindowID(uint64_t aWindowID);
+  async RemoveWindowID(uint64_t aWindowID);
+
+child:
+  async Initialize(nsString aURL);
+  async PostMessage(nsString aMessage);
+  async SetDebuggerReady(bool aReady);
+};
+
+} // namespace dom
+} // namespace mozilla